### PR TITLE
Implement is_planar()

### PR DESCRIPTION
**Pull request by merkys**
This PR implements graph planarity check in `is_planar()` for undirected graphs. The implementation is based on [left-right planarity test](https://en.wikipedia.org/w/index.php?title=Left-right_planarity_test&oldid=1158148963), whose accessable full-text (at least to me) origin seems to be [de Fraysseix and Rosenstiehl (1985)](https://doi.org/10.1007%2FBF02579375).

I have not included the mention of left-right planarity test in the code as I was unsure where it belongs. Should it be visible in the POD, or only in the code, as implementation detail?

### CODE REVIEW

**Review by mohawk2 — CHANGES_REQUESTED** [review]
Just some nits - thank you for your continuing contributions! Please could you also keep `commit --amend`ing, instead of creating several commits for one conceptual change? I believe you'd agree with me that the change isn't actually done till it's got tests and doc.

File: lib/Graph.pm
Comment (merkys) [reply lib/Graph.pm:1854]: Is it OK if I do changes to `is_bipartite()` on another branch? Or is it OK to do them in this one?

**Comment by merkys** [comment]
Thank you for such a prompt review! I will definitely try to stick to your style guidelines regarding git and whitespace usage.

File: lib/Graph.pm
Comment (mohawk2) [reply lib/Graph.pm:1854]: If it's a separate commit from the `is_planar` one, making two in total, then on this would be fine!

File: lib/Graph.pod
Comment (merkys) [reply lib/Graph.pod]: Added an implementation note with a reference in c649ecf6f164cd955246ac5cea887e58384295be.

**Comment by merkys** [comment]
I think you can squash the commits on GitHub before the merge. I probably cannot do that on my side, as this will need force-pushing (AFAIK allowed only for repository owners).

**Comment by mohawk2** [comment]
In fact, you absolutely can force-push to your own non-main branch at any time, and it is highly normal to do so. I'm happy to squash/merge this time, but please bear that in mind and force-push another time :-)

I'll release this today.

**Comment by merkys** [comment]
Oh right, I indeed can force-push to my own branch. Next time I will try to use less commits :) Thanks a lot for reviewing and merging this!

**Comment by mohawk2** [comment]
Thank you for making this module more capable! It's not a lot of effort to make new releases, but each does potentially make work for downstreams (including yourself!). This is just to request that if you *know* you have several new things you want to add, to say so on the earlier ones, in case that can reduce the number of releases in a given week ;-)

**Comment by merkys** [comment]
Yes, sorry about that. I will let you know my plans beforehand next time.

**Comment by mohawk2** [comment]
It's a nice problem for the maintainer to have! Just a thought :-)

**Pull request merged**
